### PR TITLE
docs(sn_client): Basic documentation

### PR DESCRIPTION
**Pull request by mazzi**
### Description

Adding basic documentation for `sn_client`.

Description, features, examples.

### Type of Change

Please mark the types of changes made in this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please describe):

### CODE REVIEW

File: sn_client/README.md [hunk 1]
```diff
@@ -32,6 +33,22 @@ use sn_client::Client;
...
+```bash
+$ cargo test --package sn_client --release --tests --features=local-discovery
+```
+
+## Create docs
+
+```bash
+$ cargo doc --package sn_client
+```
+
```
Comment (b-zee) [inline sn_client/README.md]: I think we can leave this out. Rust crates will usually have a `docs.rs` icon somewhere in the top of the README.

File: sn_client/README.md [hunk 2]
```diff
@@ -32,6 +33,22 @@ use sn_client::Client;
 let client = Client::new(signer, peers, req_response_timeout, custom_concurrency_limit).await?;
 ```
 
+## Running Tests
+
+Prerequisites:
+* A network running. Could be local. Refer to `safe_network/README.md` to run a local test network.
+* `SAFE_PEERS` environment variable or running the tests with --feature=local-discovery
```
Comment (b-zee) [inline sn_client/README.md]:
```suggestion
* `SAFE_PEERS` environment variable or running the tests with `--feature=local-discovery`:
```

File: sn_client/src/api.rs [hunk 3]
```diff
@@ -67,7 +67,7 @@ impl Client {
     ///
     /// Optionally specify the duration for the connection timeout.
     ///
-    /// Defaults to 180 seconds.
+    /// Defaults to `CONNECTION_TIMEOUT`.
```
Comment (b-zee) [inline sn_client/src/api.rs]: `CONNECTION_TIMEOUT` isn't a `pub` value, so this has no meaning in public API docs. Perhaps make it a `pub const` and write `[CONNECTION_TIMEOUT]` so it's hyperlinked.

File: sn_client/README.md [hunk 4]
```diff
@@ -32,6 +33,22 @@ use sn_client::Client;
 let client = Client::new(signer, peers, req_response_timeout, custom_concurrency_limit).await?;
 ```
 
+## Running Tests
+
+Prerequisites:
+* A network running. Could be local. Refer to `safe_network/README.md` to run a local test network.
```
Comment (b-zee) [inline sn_client/README.md]:
```suggestion
* A network running. Could be local. Refer to [the root `/README.md`](../README.md#using-a-local-network) to run a local test network.
```

Comment (b-zee) [reply sn_client/README.md]: @mazzi updated the suggestion

File: sn_client/src/api.rs [hunk 5]
```diff
@@ -67,7 +67,7 @@ impl Client {
     ///
     /// Optionally specify the duration for the connection timeout.
     ///
-    /// Defaults to 180 seconds.
+    /// Defaults to [`CONNECTION_TIMEOUT`].
```
Comment (b-zee) [inline sn_client/src/api.rs:70]: For this to properly link, it has to be exported by `lib.rs` too, with e.g.
```
pub use api::CONNECTION_TIMEOUT;
```
Or making the full module public by `pub mod api;`

Comment (b-zee) [reply sn_client/src/api.rs:70]: You can check quickly if the doc is proper by compiling the docs: `cargo doc --package=sn_client --open --no-deps`

**Review by maqi — CHANGES_REQUESTED** [review]
PR has 23 commits in the change history, but just 43 lines changed ?

File: sn_client/src/api.rs (hunk 5 above)
Comment (mazzi) [reply sn_client/src/api.rs:70]: I've done it with 
```rust
pub(crate) const CONNECTION_TIMEOUT: Duration = Duration::from_secs(30);
```
but still doesn't work with your command. I'll use your suggestion.

**Review by b-zee — APPROVED** [review]

**Review by maqi — APPROVED** [review]

**Pull request merged**